core/Util/plPNG.cpp

```diff
@@ -170,22 +129,37 @@ void plPNG::DecompressPNG(hsStream* S, void* buf, size_t size) {
             throw hsPNGException(__FILE__, __LINE__, "PNG output buffer is too small");
     }
 
-    png_read_image(pi.fPngReader, rows.get());
-    png_read_end(pi.fPngReader, pi.fEndInfo);
+    png_read_image(pngReader, rows.get());
+    png_read_end(pngReader, endInfo);
+    
+    png_destroy_read_struct(&pngReader, &pngInfo, &endInfo);
```

Comment:
There are a heck of a lot of lines dedicated to cleaning up this strict. I would consider using an `std::unique_ptr` to manage it. I did this with window handles in plUruLauncher